src/execution_plan/optimizations/optimizer.c

```diff
@@ -51,7 +51,8 @@ void _optimizePlan(ExecutionPlan *plan) {
 void optimizePlan(ExecutionPlan *plan) {
 	/* Handle UNION of execution plans. */
 	if(plan->is_union) {
-		for(uint i = 0; i < plan->segment_count; i++) _optimizePlan(plan->segments[i]);
```

Comment:
`segments` was a  simple heap-allocated array. I changed to `arr.h` array so we can get the length from it.
I changed it to have a uniform approach to arrays in the execution plan struct since connected componenets was `arr.h` array. It seems better and cleaner to code this way.